Docs update for MCP Apps changes since launch (#711)
apollographql/apollo-mcp-server: docs/source/mcp-apps-development.mdx

Now that the query is registered as a tool, you can run the app in ChatGPT or an MCP Apps-compatible host and prompt the LLM to show you the top products. They render using data fetched by MCP Server.

## Access tool information

> Use the `useToolInfo` hook to access the current tool name and the input the LLM passed to the tool.

Reviewer: ```suggestion
To access the tool name, as well as input passed from the LLM to the tool, use the `useToolInfo` hook.
```

}
```

`toolName` and `toolInput` are typed according to the tools defined in your application. Refer to the [typescript configuration](#typescript-configuration) section to learn how to include the generated types in your application.

<Note>

To provide specific `toolInput` types, configure a `schema` with the `apolloClientAiApps` Vite plugin. Refer to the [Vite configuration](#vite-configuration) section to learn more.

</Note>

## Populate variables from tool input

When your query includes variables, the LLM decides what values to provide for variables. This presents a problem when running the query because the variable values you provide to `useQuery` might not match the LLM provided variables.

You might consider using `useToolInfo` to populate the initial values for these variables, but this approach quickly reveals additional complexity that `useToolInfo` cannot handle on its own:

- The tool called by the agent might not be the same tool
- The input might differ from the original tool input when a user navigates away and back
- Some variables might need to synchronize with an external source, such as `props`
- Some variables might change as a result of user interaction

> The `createHydrationUtils` utility bridges this gap. Call `createHydrationUtils` with the query that provides the tool definition. This returns a `useHydratedVariables` hook used to manage variables for the query. The returned `variables` are provided to the `useQuery` hook in the `variables` option.

Reviewer: ```suggestion
The `createHydrationUtils` utility bridges the gap. Call `createHydrationUtils` with the query that provides the tool definition. Doing that returns a `useHydratedVariables` hook used to manage variables for the query. The returned `variables` are provided to the `useQuery` hook in the `variables` option.
```
